src/frequenz/lib/notebooks/reporting/data_processing.py

```diff
@@ -0,0 +1,361 @@
+structure for visualization, reporting, and analysis.
+
+Features
+--------
+- Enriches raw energy data with derived columns such as:
+  - PV production, self-consumption, feed-in, and battery charging.
+  - Net grid import and PV self-consumption share.
+- Handles time zone localization and conversion to Europe/Berlin.
+- Dynamically renames columns to more descriptive names, including
+  mapping component IDs (e.g., "PV #1", "Batterie #2").
+- Provides summary energy mix breakdowns (PV vs grid) in kWh, % share, and average kW.
+- Prepares tailored DataFrames for PV and battery analysis, supporting flexible
+  filtering by component.
+
+Main Functions
+--------------
+- `transform_energy_dataframe(df, component_types, mcfg)`:
+    Transforms a raw DataFrame with energy metrics into an enriched,
+    user-friendly format, adding PV, battery, and grid metrics.
+
+- `compute_power_df(main_df, resolution)`:
+    Computes total energy drawn from PV and grid sources over the given resolution,
+    returning a summary DataFrame with kWh, percentage, and average kW.
+
+- `print_pv_sums(main_df, resolution)`:
+    Prints total PV feed-in sums for each individual PV component
+    in a localized numeric format.
+
+- `create_pv_analysis_df(main_df, pv_filter, pvgrid_filter, pv_grid_filter_options)`:
+    Generates a DataFrame for PV analysis based on selected PV components
+    and whether to analyze PV alone, grid alone, or a grid/PV split.
+
+- `create_battery_analysis_df(main_df, bat_filter)`:
+    Creates a DataFrame for analyzing battery throughput, reshaping
+    it to long format for multi-battery analysis.
+
+Usage
+-----
+Typical usage involves:
+1. Loading a raw DataFrame with time-indexed energy measurements.
+2. Calling `transform_energy_dataframe` to process and enrich it.
+3. Using the resulting DataFrames to generate summaries,
+   for example with `compute_power_df`, `create_pv_analysis_df`, or
+   `create_battery_analysis_df` for visualization.
+"""
+
+from typing import Any, Dict, Iterable, List, Tuple, Union
+
+import pandas as pd
+
+# Constants
+TZ_NAME = "Europe/Berlin"
+COLUMN_TIMESTAMP = "timestamp"
+COLUMN_TIMESTAMP_NAMED = "Zeitpunkt"
+COLUMN_GRID = "grid"
+COLUMN_GRID_NAMED = "Netzanschluss"
+COLUMN_NET_IMPORT = "Netzbezug"
+COLUMN_CONSUMPTION = "consumption"
+COLUMN_CONSUMPTION_NAMED = "Brutto Gesamtverbrauch"
+COLUMN_BATTERY = "battery"
+COLUMN_BATTERY_POS = "battery_pos"
+COLUMN_BATTERY_NAMED = "Batterie Durchsatz"
+COLUMN_PV = "pv"
+COLUMN_PV_PROD = "PV Produktion"
+COLUMN_PV_NEG = "pv_neg"
+COLUMN_PV_EXCESS = "pv_excess"
+COLUMN_PV_FEEDIN = "PV Einspeisung"
+COLUMN_PV_SELF = "PV Eigenverbrauch"
+COLUMN_PV_BAT = "pv_bat"
+COLUMN_PV_IN_BAT = "PV in Batterie"
+COLUMN_PV_SHARE = "PV Eigenverbrauchsanteil"
+COLUMN_PV_THROUGHPUT = "PV Durchsatz"
+
+
+def transform_energy_dataframe(
+    df: pd.DataFrame,
+    component_types: List[str],
+    mcfg: Any,
```

Comment:
You shouldn't use the `Any` type annotation here. You can use the actual type, `MicrogridConfig` instead.